ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/LedgerBackedWriteService.scala

```diff
@@ -92,13 +96,15 @@ private[stores] final class LedgerBackedWriteService(ledger: Ledger, timeProvide
       FutureConverters.toJava(ledger.publishConfiguration(maxRecordTime, submissionId, config))
     }
 
-  // WriteParticipantPruningService - not supported by sandbox-classic
   override def prune(
       pruneUpToInclusive: Offset,
       submissionId: Ref.SubmissionId,
       pruneAllDivulgedContracts: Boolean,
   ): CompletionStage[state.PruningResult] =
-    CompletableFuture.completedFuture(state.PruningResult.NotPruned(Status.UNIMPLEMENTED))
+    CompletableFuture.completedFuture {
+      if (enablePruning) state.PruningResult.ParticipantPruned
```

Comment:
this is not enabling pruning, just pretending to prune, no?
how is the conformance tests pass then?
am I missing something? :thinking:

Reply:
If I understand correctly the `ApiParticipantPruningService` performs pruning in two steps:
1. pruning the write service
2. pruning the index db
For the `sandbox-classic` they are the same thing, so returning `ParticipantPruned` for the step 1. enables the step two.

Reply:
yupp, makes sense, a comment would be nice to clarify this behavior in the write service

Reply:
`pruneAllDivulgedContracts` is not meant to be supported for `sandbox-classic` because the backfiling lookup queries can start failing if used. Please return `UNIMPLEMENTED` if that flag is set